vmhost/vmhooks/managedei.go

```diff
@@ -1039,3 +1042,67 @@ func ManagedBufferToHexWithHost(host vmhost.VMHost, sourceHandle int32, destHand
 	encoded := hex.EncodeToString(mBuff)
 	managedType.SetBytes(destHandle, []byte(encoded))
 }
+
+// ManagedGetCodeMetadata VMHooks implementation.
+// @autogenerate(VMHooks)
+func (context *VMHooksImpl) ManagedGetCodeMetadata(addressHandle int32, responseHandle int32) {
+	host := context.GetVMHost()
+	ManagedGetCodeMetadataWithHost(host, addressHandle, responseHandle)
+}
+
+func ManagedGetCodeMetadataWithHost(host vmhost.VMHost, addressHandle int32, responseHandle int32) {
+	runtime := host.Runtime()
+	metering := host.Metering()
+	managedType := host.ManagedTypes()
+
+	gasToUse := metering.GasSchedule().BaseOpsAPICost.GetCodeMetadata
+	metering.UseGasAndAddTracedGas(managedGetCodeMetadataName, gasToUse)
+
+	gasToUse = metering.GasSchedule().ManagedBufferAPICost.MBufferSetBytes
+	metering.UseGasAndAddTracedGas(managedGetCodeMetadataName, gasToUse)
+
+	mBuffAddress, err := managedType.GetBytes(addressHandle)
+	if err != nil {
+		WithFaultAndHost(host, err, runtime.BaseOpsErrorShouldFailExecution())
+		return
+	}
+
+	contract, err := host.Blockchain().GetUserAccount(mBuffAddress)
+	if err != nil || check.IfNilReflect(contract) {
```

Comment:
simple checkIfNil should work. No reflect if possible.

Reply:
done